commit 01de77883140db58faed3edac982ddceea6f1b22
Author: Gregory Hainaut <gregory.hainaut@gmail.com>
Date:   Sun Jun 29 08:48:32 2014 +0200

    wx3.0: various buggy code (wx assertion)
    
    v3: reenable dc.GetSize, I didn't manage to reproduce the assertion failure

diff --git a/pcsx2/gui/Debugger/CtrlDisassemblyView.cpp b/pcsx2/gui/Debugger/CtrlDisassemblyView.cpp
index ec67f64..0d50b6f 100644
diff --git a/pcsx2/gui/Debugger/CtrlMemView.cpp b/pcsx2/gui/Debugger/CtrlMemView.cpp
index b22f541..c87c8bd 100644
diff --git a/pcsx2/gui/Debugger/DebuggerLists.cpp b/pcsx2/gui/Debugger/DebuggerLists.cpp
index 71244d3..bcfea87 100644
diff --git a/pcsx2/gui/MemoryCardFile.cpp b/pcsx2/gui/MemoryCardFile.cpp
index bc692a4..7c3cd2b 100644
--- a/pcsx2/gui/MemoryCardFile.cpp
+++ b/pcsx2/gui/MemoryCardFile.cpp
@@ -217,11 +217,13 @@ void FileMemoryCard::Close()
 {
 	for( int slot=0; slot<8; ++slot )
 	{
-		// Store checksum
-		if(!m_ispsx[slot] && !!m_file[slot].Seek(  m_chkaddr ))
-			m_file[slot].Write( &m_chksum[slot], 8 );
+		if (m_file[slot].IsOpened()) {
+			// Store checksum
+			if(!m_ispsx[slot] && !!m_file[slot].Seek(  m_chkaddr ))
+				m_file[slot].Write( &m_chksum[slot], 8 );
 
-		m_file[slot].Close();
+			m_file[slot].Close();
+		}
 	}
 }
 
diff --git a/pcsx2/gui/Panels/PluginSelectorPanel.cpp b/pcsx2/gui/Panels/PluginSelectorPanel.cpp
index d3786cf..8dfaf58 100644
--- a/pcsx2/gui/Panels/PluginSelectorPanel.cpp
+++ b/pcsx2/gui/Panels/PluginSelectorPanel.cpp
@@ -330,7 +330,8 @@ void Panels::PluginSelectorPanel::StatusPanel::SetGaugeLength( int len )
 void Panels::PluginSelectorPanel::StatusPanel::AdvanceProgress( const wxString& msg )
 {
 	m_label.SetLabel( msg );
-	m_gauge.SetValue( ++m_progress );
+	if (m_progress < m_gauge.GetRange())
+		m_gauge.SetValue( ++m_progress );
 }
 
 void Panels::PluginSelectorPanel::StatusPanel::Reset()
